code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/code_generation/DockerFileCodeGenerator.xtend

```diff
@@ -0,0 +1,157 @@
+
+import de.fhdo.lemma.model_processing.annotations.CodeGenerationModule
+import de.fhdo.lemma.model_processing.builtin_phases.code_generation.AbstractCodeGenerationModule
+import de.fhdo.lemma.model_processing.languages.LanguageDescription
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.operation.intermediate.IntermediateOperationModel
+import de.fhdo.lemma.operation.intermediate.IntermediateInfrastructureNode
+import de.fhdo.lemma.operation.intermediate.IntermediateContainer
+import de.fhdo.lemma.model_processing.code_generation.container_base.exception.DeployedServiceNumberExcpetion
+import de.fhdo.lemma.model_processing.code_generation.container_base.file.docker.DockerFile
+import de.fhdo.lemma.model_processing.code_generation.container_base.util.Util
+import java.util.Map
+import java.util.HashMap
+import java.io.File
+import org.jetbrains.annotations.NotNull
+
+
+/**
+ * Main class of the docker file generation module of the container base code generator.
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+@CodeGenerationModule(name="dockerfile")
+class DockerFileCodeGenerator extends AbstractCodeGenerationModule {
+
+    var IntermediateOperationModel model
+    var Map<String, String> content
+
+    /*
+     * Main method for the creation of the docker file
+     */
+    override execute(String[] arg0, String[] arg1) {
+        // Receive the intermediate operation model
+        model = this.intermediateModelResource.contents.get(0) as IntermediateOperationModel
+
+        // Create a Map which contains the generated source code
+        content = new HashMap
+
+        // Create a dockerfile for containers
+        model.containers.forEach[container |
+            this.createDockerFileForContainer(container)
+        ]
+
+        // Create dockerfile for infrastructure nodes
+        model.infrastructureNodes.forEach[node |
+            val property = node.defaultValues.findFirst[property |
+                property.technologySpecificProperty.name == "imageType"
+            ]
+
+            if (property.value.toLowerCase != "dockerhub")
+                this.createDockerFileForInfrastructureNode(node)
+        ]
+
+        println("CodeGenerationModule dockerfile finished.")
+        return withCharset(content, "UTF-8");
+    }
+
+    /*
+     * Receive the language description for the intermediate operation model package.
+     */
+    @NotNull
+    override getLanguageDescription() {
+        new LanguageDescription(IntermediatePackage.eINSTANCE)
+    }
+
+    /*
+     * This method is responsible for the creation of the dockerfile.
+     * The creation of the dockerfile is only supported for container deploying a single service,
+     * because of an architectural point of view the deployment of more than one microservice in a
+     * container breaks the principle of an independent deployment
```

Comment:
End with period.

Reply:
In addition: LEMMA does by intent enable the deployment of more than one service to a container. If the modeler decides to configure a container for multi-deployments is her (hopefully conscious) decision. So no need to constrain the deployment to one service here.